aws/resource_aws_wafv2_rule_group.go

```diff
@@ -0,0 +1,1346 @@
+		m["type"] = aws.StringValue(t.Type)
+		out[i] = m
+	}
+	return out
+}
+
+func flattenWafv2IpSetReferenceStatement(i *wafv2.IPSetReferenceStatement) interface{} {
+	if i == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"arn": aws.StringValue(i.ARN),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2GeoMatchStatement(g *wafv2.GeoMatchStatement) interface{} {
+	if g == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"country_codes": flattenStringList(g.CountryCodes),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2NotStatement(a *wafv2.NotStatement) interface{} {
+	if a == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"statement": []interface{}{flattenWafv2Statement(a.Statement)},
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2OrStatement(a *wafv2.OrStatement) interface{} {
+	if a == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"statement": flattenWafv2Statements(a.Statements),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2RegexPatternSetReferenceStatement(r *wafv2.RegexPatternSetReferenceStatement) interface{} {
+	if r == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"arn":                 aws.StringValue(r.ARN),
+		"field_to_match":      flattenWafv2FieldToMatch(r.FieldToMatch),
+		"text_transformation": flattenWafv2TextTransformations(r.TextTransformations),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2SizeConstraintStatement(s *wafv2.SizeConstraintStatement) interface{} {
+	if s == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"comparison_operator": aws.StringValue(s.ComparisonOperator),
+		"field_to_match":      flattenWafv2FieldToMatch(s.FieldToMatch),
+		"size":                int(aws.Int64Value(s.Size)),
+		"text_transformation": flattenWafv2TextTransformations(s.TextTransformations),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2SqliMatchStatement(s *wafv2.SqliMatchStatement) interface{} {
+	if s == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"field_to_match":      flattenWafv2FieldToMatch(s.FieldToMatch),
+		"text_transformation": flattenWafv2TextTransformations(s.TextTransformations),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2XssMatchStatement(s *wafv2.XssMatchStatement) interface{} {
+	if s == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"field_to_match":      flattenWafv2FieldToMatch(s.FieldToMatch),
+		"text_transformation": flattenWafv2TextTransformations(s.TextTransformations),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2VisibilityConfig(config *wafv2.VisibilityConfig) interface{} {
+	if config == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"cloudwatch_metrics_enabled": aws.BoolValue(config.CloudWatchMetricsEnabled),
+		"metric_name":                aws.String(*config.MetricName),
```

Comment:
to keep everything in this map as values, let's use the string value with `aws.StringValue(config.MetricName)` instead